drivers/usb/device/usb_dc_nrf5.c

```diff
@@ -1732,14 +1732,14 @@ int usb_dc_attach(void)
 	usbd_enable_interrupts();
 
 	ret = hf_clock_enable(true, false);
-	if (ret) {
+	if (ret && ret != -EINPROGRESS) {
 		goto err_clk_enable;
 	}
 
 	endpoint_ctx_init();
 	ctx->attached = true;
 
-	return ret;
+	return 0;
```

Comment:
Should -EINPROGRESS be returned here too?

Reply:
I think that -EINPROGRESS could be hidden from user here.
The resulting state is the same whether start operation was already in progress or not.

Reply:
I can see other problem with this solution - it is laud - I mean, the hf_clock_enable is called, and inside this function errors are printed even if it is ok for the driver to return -EINPROGRESS.

Reply:
I did not understand, could you ellaborate on the errors?

Reply:
The `usb_dc_attach` function calls hf_clock_enable and ignores the error if it gets `-EINPROGRESS` - that part seems ok. But look inside `hf_clock_enable` function:

```c
...
	if (on) {
		ret = clock_control_on(clock, (void *)blocking);
	} else {
		ret = clock_control_off(clock, (void *)blocking);
	}

	if (ret) {
		SYS_LOG_ERR("NRF5 HF clock %s fail: %d",
			    on ? "start" : "stop", ret);
		return ret;
	}
...
```

In this part the error information is generated. So the solution is noisy as you get error message printed even if it is ignored in the caller.

Reply:
Thanks for catching it... let me see what I can do, meanwhile please put the PR in request for changes.

Reply:
Done.

Reply:
@pawelzadrozniak  Updated the PR, please review.